### ISSUE TITLE
Smart forum links with preview

### ISSUE THREAD

**Issue by traumschule** [Joystream/pioneer#4438]
Linking to a forum thread requires inserting a full link or manually crafting a CKEditor link with title and relative link `/forum/..` to be valid across deployments.

0. [ ] Define a syntax thread links are stored on-chain.

## Editor
1. [ ] When user enters `#` offer a list of threads to choose from, filtered by ID.
2. [ ]  When a full link is inserted expand thread title to click on.

![gh-forum](https://github.com/Joystream/pioneer/assets/31551045/837dbf3b-8ad3-40eb-ab10-1c17e33d1e53)

# Display Post
3. [ ] Show thread links with ID and title as link: `#427 Leads Scoring And Effects On Salary` 
4. [ ] On hover (tooltip) show abbreviated preview of first post similar to GH.
5. [ ] On click navigate to the thread in same instance / tab.

Related: #4169

**Comment by dmtrjsg** [comment]
@traumschule 

Looks like it was already in place but doesn't fully work: 
<img width="416" alt="Screenshot 2023-06-23 at 14 22 33" src="https://github.com/Joystream/pioneer/assets/93924525/b5319a7c-bf3b-4d6f-8d95-945602d88a88">

**Comment by traumschule** [comment]
Yes it was always a bit unclear :) https://github.com/Joystream/pioneer/issues/3093
